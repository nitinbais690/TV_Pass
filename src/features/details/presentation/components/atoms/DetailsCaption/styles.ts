import { appFonts } from 'core/styles/AppStyles';
import { StyleSheet } from 'react-native';

export const styles = (appColors: any) => {
    return StyleSheet.create({
        titleStyle: {
            color: appColors.secondary,
            opacity: 0.3,
            fontFamily: appFonts.semibold,
        },
    });
};
