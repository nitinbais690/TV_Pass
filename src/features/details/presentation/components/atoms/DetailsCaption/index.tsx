import { appFontStyle } from 'core/styles/AppStyles';
import React from 'react';
import { Text } from 'react-native';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import { styles } from './styles';

export default function DetailsCaption(props: { name: string | undefined; style?: {} }) {
    let titleStyles = styles(useAppColors());
    let fontStyle = appFontStyle;

    return (
        <Text style={[fontStyle.sublineText, titleStyles.titleStyle, props.style]}>
            {props.name ? props.name : 'A'}
        </Text>
    );
}
