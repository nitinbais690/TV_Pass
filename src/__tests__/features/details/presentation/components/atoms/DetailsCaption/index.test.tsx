import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import { StyleSheet } from 'react-native';
import DetailsCaption from 'features/details/presentation/components/atoms/DetailsCaption';

jest.mock('core/presentation/hooks/use-app-colors');

describe('detailsCaption', () => {
    const text: string = 'Inception';

    test('render with default style', () => {
        const { container, queryByText } = render(<DetailsCaption name={text} />);
        expect(queryByText(text)).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with custom style', () => {
        const styles = StyleSheet.create({
            upperCase: {
                textTransform: 'uppercase',
            },
        });

        const { container, queryByText } = render(<DetailsCaption name={text} style={styles.upperCase} />);
        expect(queryByText(text)).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });

    test('render with empty text', () => {
        const { container } = render(<DetailsCaption name={''} />);
        expect(container).toBeDefined();
        expect(toJSON(container)).toMatchSnapshot();
    });
});
