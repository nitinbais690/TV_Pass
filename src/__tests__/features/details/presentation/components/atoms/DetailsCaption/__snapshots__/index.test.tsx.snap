// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detailsCaption render with custom style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 14,
          "fontWeight": "600",
        },
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Semibold",
          "opacity": 0.3,
        },
        Object {
          "textTransform": "uppercase",
        },
      ]
    }
  >
    Inception
  </Text>
</View>
`;

exports[`detailsCaption render with default style 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 14,
          "fontWeight": "600",
        },
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Semibold",
          "opacity": 0.3,
        },
        undefined,
      ]
    }
  >
    Inception
  </Text>
</View>
`;

exports[`detailsCaption render with empty text 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 14,
          "fontWeight": "600",
        },
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Semibold",
          "opacity": 0.3,
        },
        undefined,
      ]
    }
  >
    A
  </Text>
</View>
`;
